lib/elixir/src/elixir_module.erl

```diff
@@ -70,16 +70,21 @@ compile(Line, Module, Block, Vars, E) ->
 
   try
     put_compiler_modules([Module | CompilerModules]),
-    {Result, NE} = eval_form(Line, Module, Data, Block, Vars, E),
+    {Result, NE, OverridablePairsNotOverridden} = eval_form(Line, Module, Data, Block, Vars, E),
 
     PersistedAttributes = ets:lookup_element(Data, ?persisted_attr, 2),
     Attributes = attributes(Line, File, Data, PersistedAttributes),
     OnLoad = ets:lookup_element(Data, 'on_load', 2),
     [elixir_locals:record_local(Tuple, Module) || Tuple <- OnLoad],
 
     {AllDefinitions, Unreachable} = elixir_def:fetch_definitions(File, Module),
+
+    %% Ignore functions that do not override the original since these are
+    %% checked instead in `Module.make_overridable/2`
+    DefinitionsToCheck = [Definition || Definition = {Pair, _, _, _} <- AllDefinitions, not lists:member(Pair, OverridablePairsNotOverridden)],
```

Comment:
Don't do this check here. Pass all definitions as well as all overridable pairs to `check_behaviours_and_impls` and there you can check if it needs to be checked or not. The reason is that we want to add more checks to `check_behaviours_and_impls` in the future so we need to have all definitions there.